flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -283,7 +283,7 @@ public Task(
 		int targetSlotNumber,
 		MemoryManager memManager,
 		IOManager ioManager,
-		ShuffleEnvironment<?, ?> shuffleEnvironment,
+		JobAwareShuffleEnvironment<?, ?> shuffleEnvironment,
```

Comment:
We could still reference `ShuffleEnvironment` here if making `JobAwareShuffleEnvironmentImpl` implements both `JobAwareShuffleEnvironment` and `ShuffleEnvironment` as mentioned above.